py/server/deephaven/rollup_table.py

```diff
@@ -0,0 +1,195 @@
+
+    def where(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.where` operation recorded to be
+        applied to nodes."""
+        filters = to_sequence(filters)
+        j_filter_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFilterOperationsRecorder)
+        return self.__class__(j_filter_ops_recorder.where(filters))
+
+
+class RollupNodeOperationsRecorder(JObjectWrapper, FormatOperationsRecorder,
+                                   SortOperationsRecorder):
+    """Recorder for node-level operations to be applied when gathering snapshots of RollupTable. Supported operations
+    include column formatting and sorting.
+
+    Note: It should not be instantiated directly. User code must call :meth:`~RollupTable.node_operation_recorder` to
+    create an instance of the recorder.
+    """
+
+    j_object_type = _JNodeOperationsRecorder
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_node_ops_recorder
+
+    def __init__(self, j_node_ops_recorder: jpy.JType):
+        self.j_node_ops_recorder = j_node_ops_recorder
+
+
+class RollupTable(JObjectWrapper):
+    """ A RollupTable is generated as a result of applying the :meth:`~deephaven.table.Table.rollup` operation on a
+    :class:`~deephaven.table.Table`.
+
+    Note: RollupTable should not be instantiated directly by user code.
+    """
+    j_object_type = _JRollupTable
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_rollup_table
+
+    def __init__(self, j_rollup_table: jpy.JType, aggs: Sequence[Aggregation], include_constituents: bool,
+                 by: Sequence[str]):
+        self.j_rollup_table = j_rollup_table
+        self.aggs = aggs
+        self.include_constituents = include_constituents
+        self.by = by
+
+    def node_operation_recorder(self, node_type: NodeType) -> RollupNodeOperationsRecorder:
+        """Creates a RollupNodeOperationsRecorder for per-node operations to apply during snapshotting of this
+        RollupTable. The recorded node operations will be applied only to the node of the provided NodeType. See
+        :class:`NodeType` for details.
+
+        Args:
+            node_type (NodeType): the type of node tables that the recorded operations will be applied to; if it is
+                :attr:`NodeType.CONSTITUENT`, the RollupTable must be created with include_constituents=True.
+
+        Returns:
+            a RollupNodeOperationsRecorder
+
+        Raises:
+            DHError
+        """
+        try:
+            return RollupNodeOperationsRecorder(j_node_ops_recorder=self.j_rollup_table.makeNodeOperationsRecorder(
+                node_type.value))
+        except Exception as e:
+            raise DHError(e, "failed to create a RollupNodeOperationsRecorder.") from e
+
+    def with_node_operations(self, node_ops_recorders: List[RollupNodeOperationsRecorder]) -> RollupTable:
```

Comment:
Is there a reasonable but shorter alternative to `node_ops_recorders` (e.g. `recorders`)?